common/src/test/java/com/revenuecat/purchases/backend_integration_tests/BaseBackendIntegrationTest.kt

```diff
@@ -0,0 +1,102 @@
+// common module build.gradle and change the API KEY in Constants.kt
+@RunWith(AndroidJUnit4::class)
+abstract class BaseBackendIntegrationTest {
+
+    companion object {
+        private val TIMEOUT = 5.seconds
+
+        @BeforeClass
+        @JvmStatic
+        fun setupClass() {
+            if (!canRunIntegrationTests()) {
+                error("You need to set required constants in Constants.kt")
+            }
+        }
+
+        private fun canRunIntegrationTests() = Constants.apiKey != "REVENUECAT_API_KEY"
+            && Constants.loadShedderApiKey != "LOAD_SHEDDER_API_KEY"
+    }
+
+    lateinit var appConfig: AppConfig
+    lateinit var dispatcher: Dispatcher
+    lateinit var diagnosticsDispatcher: Dispatcher
+    lateinit var sharedPreferences: SharedPreferences
+    lateinit var sharedPreferencesEditor: SharedPreferences.Editor
+    lateinit var eTagManager: ETagManager
+    lateinit var signingManager: SigningManager
+    lateinit var httpClient: HTTPClient
+    lateinit var backendHelper: BackendHelper
+
+    lateinit var backend: Backend
+
+    @Before
+    fun setUp() {
+        appConfig = mockk<AppConfig>().apply {
+            every { baseURL } returns URL("https://api.revenuecat.com")
+            every { store } returns Store.PLAY_STORE
+            every { platformInfo } returns PlatformInfo("test-flavor", version = null)
+            every { languageTag } returns "en-US"
+            every { versionName } returns "test-version-name"
+            every { packageName } returns "com.revenuecat.purchases.backend_tests"
+            every { finishTransactions } returns true
+        }
+        dispatcher = Dispatcher(Executors.newSingleThreadScheduledExecutor())
+        diagnosticsDispatcher = Dispatcher(Executors.newSingleThreadScheduledExecutor())
+        sharedPreferencesEditor = mockk<SharedPreferences.Editor>().apply {
+            every { putString(any(), any()) } returns this
+            every { apply() } just Runs
+        }
+        sharedPreferences = mockk<SharedPreferences>().apply {
+            every { getString(any(), any()) } answers { secondArg() as String? }
+            every { edit() } returns sharedPreferencesEditor
+        }
+        eTagManager = ETagManager(sharedPreferences)
+        signingManager = SigningManager(SignatureVerificationMode.Disabled)
+        httpClient = HTTPClient(appConfig, eTagManager, diagnosticsTrackerIfEnabled = null, signingManager)
+        backendHelper = BackendHelper(apiKey(), dispatcher, appConfig, httpClient)
+        backend = Backend(appConfig, dispatcher, diagnosticsDispatcher, httpClient, backendHelper)
+    }
+
+    abstract fun apiKey(): String
+
+    protected fun ensureBlockFinishes(block: (CountDownLatch) -> Unit) {
+        val latch = CountDownLatch(1)
+        block(latch)
+        latch.await(TIMEOUT.inWholeSeconds, TimeUnit.SECONDS)
+        assertThat(latch.count).isEqualTo(0)
+    }
```

Comment:
I love this!

Reply:
Yeah, I think this could be useful for many other tests as well. For now I'm just using it here but we could consider it moving to the test-utils module and use it in many other tests.